extensions/amp-render/1.0/test/test-amp-render.js

```diff
@@ -0,0 +1,114 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../../../amp-bind/0.1/amp-bind';
+import '../../../amp-mustache/0.2/amp-mustache';
+import '../amp-render';
+import {htmlFor} from '../../../../src/static-template';
+import {toggleExperiment} from '../../../../src/experiments';
+import {waitFor} from '../../../../testing/test-helper';
+import {whenUpgradedToCustomElement} from '../../../../src/dom';
+
+describes.realWin(
+  'amp-render-v1.0',
+  {
+    amp: {
+      extensions: ['amp-mustache:0.2', 'amp-bind:0.1', 'amp-render:1.0'],
+    },
+  },
+  (env) => {
+    let win, doc, html, element;
+
+    async function waitRendered() {
+      await whenUpgradedToCustomElement(element);
+      await element.buildInternal();
+      await waitFor(() => {
+        // The rendered container inserts a <div> element.
+        const div = element.querySelector('div');
+        return div && div.textContent;
+      }, 'wrapper div rendered');
+      return element.querySelector('div');
+    }
+
+    async function getRenderedData() {
+      const wrapper = await waitRendered();
+      return wrapper.textContent;
+    }
+
+    beforeEach(async function () {
+      win = env.win;
+      doc = win.document;
+      html = htmlFor(doc);
+      toggleExperiment(win, 'amp-render', true, true);
+    });
+
+    it('renders from amp-state', async () => {
+      const ampState = html`
+        <amp-state id="theFood">
+          <script type="application/json">
+            {
+              "name": "Bill"
+            }
+          </script>
+        </amp-state>
+      `;
+      doc.body.appendChild(ampState);
+
+      element = html`
+        <amp-render
+          src="amp-state:theFood"
+          width="auto"
+          height="140"
+          layout="fixed-height"
+        >
+          <template type="amp-mustache"><p>Hello {{name}}</p></template>
+        </amp-render>
+      `;
+      doc.body.appendChild(element);
+
+      await whenUpgradedToCustomElement(ampState);
+      await ampState.buildInternal();
+
+      const text = await getRenderedData();
+      expect(text).to.contain('Hello Bill');
+    });
+
+    it('renders json from src', async () => {
+      element = html`
+        <amp-render
+          src="https://example.com/data.json"
+          width="auto"
+          height="140"
+          layout="fixed-height"
+        >
+          <template type="amp-mustache"><p>Hello {{name}}</p></template>
+        </amp-render>
+      `;
+      doc.body.appendChild(element);
+
+      const ampRender = await element.getImpl(false);
+
+      const fetchFn = () => {
+        return Promise.resolve({name: 'Joe'});
+      };
+
+      env.sandbox.stub(ampRender, 'getJsonFn_').returns(fetchFn);
+
+      const text = await getRenderedData();
+      expect(text).to.contain('Hello Joe');
```

Comment:
Are these using `contain` instead of `equal` due to white space? If so, can we `trim` first to assert `equal`? It shouldn't render anything else.

Reply:
`contain` was a remnant of copy/paste. Changed to `equal`.